segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerEventProcessorImpl.java

```diff
@@ -288,7 +288,8 @@ public EventProcessorImpl(@NonNull String name, int containerId, @NonNull Direct
             this.closed = new AtomicBoolean(false);
             this.failedIteration = new AtomicBoolean(false);
             // Set with actual value when upon initialization.
-            this.segmentStartOffset = new AtomicLong(segment.getInfo().getStartOffset());
+            this.processedUpToOffset = new AtomicLong(segment.getInfo().getStartOffset());
+            this.lastTruncationOffset = new AtomicLong(0);
```

Comment:
should this be getStartOffset as well ? 
otherwise on first iteration getOutstandingBytes will give wrong answer and shouldTruncate will be true

Reply:
Yes, good catch. The first iteration after a container restart, this would be normally truncate the Segment (after that first iteration, things would work again)